superset-frontend/src/components/DatabaseSelector/index.tsx

```diff
@@ -97,193 +89,189 @@ export default function DatabaseSelector({
   schema,
   sqlLabMode = false,
 }: DatabaseSelectorProps) {
-  const [currentDbId, setCurrentDbId] = useState(dbId);
-  const [currentSchema, setCurrentSchema] = useState<string | undefined>(
-    schema,
+  const [currentDb, setCurrentDb] = useState(
+    db
+      ? { label: `${db.backend}: ${db.database_name}`, value: db.id }
+      : undefined,
+  );
+  const [currentSchema, setCurrentSchema] = useState<SchemaValue | undefined>(
+    schema ? { label: schema, value: schema } : undefined,
   );
-  const [schemaLoading, setSchemaLoading] = useState(false);
-  const [schemaOptions, setSchemaOptions] = useState([]);
+  const [refresh, setRefresh] = useState(0);
 
-  function fetchSchemas(databaseId: number, forceRefresh = false) {
-    const actualDbId = databaseId || dbId;
-    if (actualDbId) {
-      setSchemaLoading(true);
-      const queryParams = rison.encode({
-        force: Boolean(forceRefresh),
-      });
-      const endpoint = `/api/v1/database/${actualDbId}/schemas/?q=${queryParams}`;
-      return SupersetClient.get({ endpoint })
-        .then(({ json }) => {
+  const loadSchemas = useMemo(
+    () => async (): Promise<{
+      data: SchemaValue[];
+      totalCount: number;
+    }> => {
+      if (currentDb) {
+        const queryParams = rison.encode({ force: refresh > 0 });
+        const endpoint = `/api/v1/database/${currentDb.value}/schemas/?q=${queryParams}`;
+
+        // The endpoint currently does not support pagination
+        return SupersetClient.get({ endpoint }).then(({ json }) => {
```

Comment:
Could potentially change this to a `TODO:` if you have to push commits to the PR (would be nice to add pagination in a follow-up)